frontend/src/app/main/data/workspace/persistence.cljs

```diff
@@ -71,93 +70,89 @@
               ;; Disable reload stoper
               (swap! st/ongoing-tasks disj :workspace-change)
               (st/emit! (update-persistence-status {:status :saved})))]
-        (->> (rx/merge
-              (->> stream
-                   (rx/filter dch/commit-changes?)
-                   (rx/map deref)
-                   (rx/filter local-file?)
-                   (rx/tap on-dirty)
-                   (rx/buffer-until notifier)
-                   (rx/filter (complement empty?))
-                   (rx/map (fn [buf]
-                             (->> (into [] (comp (map #(assoc % :id (uuid/next)))
-                                                 (map #(assoc % :file-id file-id)))
-                                        buf)
-                                  (persist-changes file-id))))
-                   (rx/tap on-saving)
-                   (rx/take-until (rx/delay 100 stoper)))
-              (->> stream
-                   (rx/filter dch/commit-changes?)
-                   (rx/map deref)
-                   (rx/filter library-file?)
-                   (rx/filter (complement #(empty? (:changes %))))
-                   (rx/map persist-synchronous-changes)
-                   (rx/take-until (rx/delay 100 stoper)))
-              (->> stream
-                   (rx/filter (ptk/type? ::changes-persisted))
-                   (rx/tap on-saved)
-                   (rx/ignore)
-                   (rx/take-until stoper)))
-             (rx/subs #(st/emit! %)
-                      (constantly nil)
-                      (fn []
-                        (on-saved))))))))
+
+        (rx/merge
+         (->> stream
+              (rx/filter dch/commit-changes?)
+              (rx/map deref)
+              (rx/filter local-file?)
+              (rx/tap on-dirty)
+              (rx/filter (complement empty?))
+              (rx/map (fn [commit]
+                        (-> commit
+                            (assoc :id (uuid/next))
+                            (assoc :file-id file-id))))
+              (rx/observe-on :async)
+              (rx/tap #(swap! commits conj %))
+              (rx/take-until (rx/delay 100 stoper))
+              (rx/finalize (fn []
+                             (log/debug :hint "finalize persistence: changes watcher"))))
+
+         (->> (rx/from-atom commits)
+              (rx/filter (complement empty?))
+              (rx/sample-when (rx/merge
+                               (rx/interval 5000)
```

Comment:
Why is this necessary? Won't it trigger a save process even when there is no commits to save?